helpers/common.go

```diff
@@ -72,26 +83,44 @@ func UnameMachine() (string, error) {
 // 4.18.0-305.7.1.el8_4.centos.x86_64 (centos)
 // 4.18.0-305.7.1.el8_4.centos.plus.x86_64 (centos + plus repo)
 // 5.13.13-arch1-1 (archlinux)
-//
-func CompareKernelRelease(base, given string) int {
+func CompareKernelRelease(base, given string) (int, error) {
 	b := strings.Split(base, "-") // [base]-xxx
 	b = strings.Split(b[0], ".")  // [major][minor][patch]
 
+	if len(b) == 2 {
+		b = append(b, "0")
+	}
+	if len(b) != 3 {
+		return KernelVersionInvalid, fmt.Errorf("invalid base kernel version format: %s", base)
+	}
+
 	g := strings.Split(given, "-")
 	g = strings.Split(g[0], ".")
+	if len(g) == 2 {
+		g = append(b, "0")
+	}
+	if len(g) != 3 {
+		return KernelVersionInvalid, fmt.Errorf("invalid given kernel version format: %s", given)
+	}
 
 	for n := 0; n <= 2; n++ {
-		i, _ := strconv.Atoi(g[n])
-		j, _ := strconv.Atoi(b[n])
+		i, err := strconv.Atoi(g[n])
+		if err != nil {
+			return KernelVersionInvalid, fmt.Errorf("invalid given kernel version value: %s", g[n])
+		}
```

Comment:
Perhaps returning `given` plus `g[n]` together with strconv error? It would provide more information for the consumer.